### PR TITLE
resync from typeshed

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

**Comment by graingert** [comment]
@ncoghlan ^

File: contextlib2/_typeshed.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from typing import TypeVar
```
Comment (graingert) [inline contextlib2/_typeshed.py:1]:
```suggestion
from typing import TypeVar  #pragma no cover
```

File: contextlib2/_typeshed.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+# Use for "self" annotations:
+#   def __enter__(self: Self) -> Self: ...
+Self = TypeVar("Self")
```
Comment (graingert) [inline contextlib2/_typeshed.py:5]:
```suggestion
Self = TypeVar("Self")  # pragma: no cover
```

**Comment by adamchainz** [comment]
This LGTM

**Pull request merged**